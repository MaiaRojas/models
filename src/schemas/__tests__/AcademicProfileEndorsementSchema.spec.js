const mongoose = require('mongoose/browser');
const AcademicProfileEndorsementSchema = require('../AcademicProfileEndorsementSchema')(mongoose);
const EndorsementSchema = require('../EndorsementSchema')(mongoose);
const UserSchema = require('../UserSchema')(mongoose);

describe('AcademicProfileEndorsementSchema', () => {
  it('should fail validation when missing fields are provided', () => {
    const doc = new mongoose.Document({}, AcademicProfileEndorsementSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should successfully validate with proper values', (done) => {
    const endorsement = new mongoose.Document({}, EndorsementSchema);
    const createdBy = new mongoose.Document({
      uid: 'xxxxxxxxxx',
      email: 'foo@bar.baz',
    }, UserSchema);

    const doc = new mongoose.Document({
      createdBy: createdBy._id,
      endorsement: endorsement._id,
    }, AcademicProfileEndorsementSchema);

    doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
