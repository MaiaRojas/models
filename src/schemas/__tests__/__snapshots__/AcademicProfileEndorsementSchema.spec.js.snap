// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcademicProfileEndorsementSchema should fail validation when missing fields are provided 1`] = `
Object {
  "createdBy": [ValidatorError: Path \`createdBy\` is required.],
  "endorsement": [ValidatorError: Path \`endorsement\` is required.],
}
`;
